Removing Multiplayer Component Causes Build Error

If you remove an auto-component XML, AZCodeGen won't remove the generated code file and it causes a compile error.
Users has to manually find and delete the generated file inside the build folder.

**Steps to reproduce**
1. Create a new multiplayer autocomponent xml and add it to cmake files. (Set ComponentOverride and ControllerOverride to false, so you dont need any custom code file)
2. Compile (see that compile works)
3. Delete the the autocomponent and remove it from cmake file list.
4. Compile

**Expected behavior**
Code compiles successfully.

**Actual behavior**
Build error occurs. User has to manually delete the generated code file and rebuild

```
Build started...
5>------ Build started: Project: Multiplayer.Static, Configuration: profile x64 ------
5>unity_2_cxx.cxx
5>Multiplayer.Static.vcxproj -> D:\prj\MultiplayerSample\build\windows\lib\profile\Multiplayer.Static.lib
6>------ Build started: Project: MultiplayerSample.Static, Configuration: profile x64 ------
7>------ Build started: Project: Multiplayer.Debug, Configuration: profile x64 ------
8>------ Build started: Project: Multiplayer (Gems\Multiplayer\Code\Multiplayer), Configuration: profile x64 ------
6>Running AutoGen for MultiplayerSample.Static
6>*.AutoComponent.xml,AutoComponent_Header.jinja,$path/$fileprefix.AutoComponent.h : error Processing expansion rule
6>An unexpected error occurred, please report the error you encountered and include your build output
6>Traceback (most recent call last):
6>  File "D:\o3de\cmake\AzAutoGen.py", line 130, in ProcessTemplateConversion
6>    xml = etree.parse(dataInputFile)
6>  File "D:\o3de\python\runtime\python-3.10.5-rev1-windows\python\lib\xml\etree\ElementTree.py", line 1224, in parse
6>    tree.parse(source, parser)
6>  File "D:\o3de\python\runtime\python-3.10.5-rev1-windows\python\lib\xml\etree\ElementTree.py", line 569, in parse
6>    source = open(source, "rb")
6>FileNotFoundError: [Errno 2] No such file or directory: 'D:\\prj\\MultiplayerSample\\Gem\\Code\\Source\\AutoGen\\SomeNewComponent.AutoComponent.xml'
6>
6>During handling of the above exception, another exception occurred:
6>
6>Traceback (most recent call last):
6>  File "D:\o3de\cmake\AzAutoGen.py", line 419, in <module>
6>    autoGenResult = ExecuteExpansionRules(autogenConfig, dataInputSet, outputFiles)
6>  File "D:\o3de\cmake\AzAutoGen.py", line 374, in ExecuteExpansionRules
6>    ProcessExpansionRule(autogenConfig, sourceFiles, templateFiles, templateCache, expansionRule, dataInputSet, outputFiles)
6>  File "D:\o3de\cmake\AzAutoGen.py", line 332, in ProcessExpansionRule
6>    ProcessTemplateConversion(autogenConfig, dataInputSet, dataInputFiles, templateFile, outputFileAbsolute, templateCache)
6>  File "D:\o3de\cmake\AzAutoGen.py", line 167, in ProcessTemplateConversion
6>    PrintError('%s(%s) : %s' % (fileinput.filename(), str(fileinput.filelineno()), e.message))
6>  File "D:\o3de\python\runtime\python-3.10.5-rev1-windows\python\lib\fileinput.py", line 121, in filename
6>    raise RuntimeError("no active input()")
6>RuntimeError: no active input()
6>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'D:\prj\MultiplayerSample\build\windows\CMakeFiles\c89cbd3be96997261947eb8d2de4f388\SomeNewComponent.AutoComponent.h.rule;D:\prj\MultiplayerSample\Gem\Code\CMakeLists.txt' exited with code 1.
6>Done building project "MultiplayerSample.Static.vcxproj" -- FAILED.
9>------ Build started: Project: MultiplayerSample (Gem\Code\MultiplayerSample), Configuration: profile x64 ------
9>Building Custom Rule D:/prj/MultiplayerSample/Gem/Code/CMakeLists.txt
9>CMake is re-running because D:/prj/MultiplayerSample/build/windows/External/MultiplayerSample-512ed798/Gem/Code/CMakeFiles/generate.stamp is out-of-date.
9>  the file 'D:/prj/MultiplayerSample/Gem/Code/multiplayersample_files.cmake'
9>  is newer than 'D:/prj/MultiplayerSample/build/windows/External/MultiplayerSample-512ed798/Gem/Code/CMakeFiles/generate.stamp.depend'
9>  result='-1'
9>-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
9>-- Using package D:/.o3de/3rdParty/packages/zlib-1.2.11-rev5-windows
9>-- Configuring done
9>-- Generating done
9>-- Build files have been written to: D:/prj/MultiplayerSample/build/windows
9>   Creating library D:/prj/MultiplayerSample/build/windows/lib/profile/MultiplayerSample.lib and object D:/prj/MultiplayerSample/build/windows/lib/profile/MultiplayerSample.exp
9>MultiplayerSample.vcxproj -> D:\prj\MultiplayerSample\build\windows\bin\profile\MultiplayerSample.dll
10>------ Build started: Project: MultiplayerSample.ServerLauncher, Configuration: profile x64 ------
10>Copying MultiplayerSample.ServerLauncher runtime dependencies to output...
10>MultiplayerSample.ServerLauncher.vcxproj -> D:\prj\MultiplayerSample\build\windows\bin\profile\MultiplayerSample.ServerLauncher.exe
========== Build: 9 succeeded, 1 failed, 112 up-to-date, 0 skipped ==========
```

**Found in Branch**
development latest October 28 2022

**Desktop/Device (please complete the following information):**
 Windows VS 2019

--- a/cmake/LyAutoGen.cmake
+++ b/cmake/LyAutoGen.cmake
@@ -39,6 +39,7 @@ function(ly_add_autogen)
         string(STRIP "${AUTOGEN_OUTPUTS}" AUTOGEN_OUTPUTS)
         set(AZCG_DEPENDENCIES ${AZCG_INPUTFILES})
         list(APPEND AZCG_DEPENDENCIES "${LY_ROOT_FOLDER}/cmake/AzAutoGen.py")
+        set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${AZCG_DEPENDENCIES})
         add_custom_command(
             OUTPUT ${AUTOGEN_OUTPUTS}
             DEPENDS ${AZCG_DEPENDENCIES}

